torch/csrc/jit/frontend/script_type_parser.cpp

```diff
@@ -209,9 +209,19 @@ c10::optional<std::string> ScriptTypeParser::parseBaseTypeName(
     case '.': {
       auto select = Select(expr);
       const std::string& name = select.selector().name();
-      // Special case for torch.Tensor
-      if (isTorch(select.value()) && name == "Tensor") {
-        return "Tensor";
+      // Special case for torch.Tensor and its' subclasses
+      const std::unordered_set<std::string> tensor_subtypes = {
+          "Tensor",
+          "LongTensor",
```

Comment:
"DoubleTensor" is missing from this list. Is it intended?

Reply:
done